File: 백준/Silver/10828. 스택/스택.py
import sys
input = sys.stdin.readline

stack = []

n = int(input().strip())

for _ in range(n):
    cmd = input().strip().split()

    if cmd[0] == "push":
        stack.append(int(cmd[1]))
    elif cmd[0] == "pop":
        if stack:
            print(stack.pop())
        else:
            print(-1)
    elif cmd[0] == "size":
        print(len(stack))
    elif cmd[0] == "empty":
        print(0 if stack else 1)
    elif cmd[0] == "top":
        if stack:
            print(stack[-1])
        else:
            print(-1)

File: 백준/Silver/10828. 스택/스택.cc
#include<iostream>
#include<stack>
using namespace std;

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

    int x;
    cin >> x;

    stack<int> s;

    while (x--) {
        string cmd;
        cin >> cmd;
        if(cmd == "push") {
            int num;
            cin >> num;
            s.push(num);
        } else if(cmd == "pop") {
            if(!s.empty()) {
                cout << s.top() << "\n";
                s.pop();
            } else {
                cout << "-1\n";
            }
        } else if(cmd == "top") {
            if(!s.empty()) {
                cout << s.top() << "\n";
            } else {
                cout << "-1\n";
            }
        } else if(cmd == "size") {
            cout << s.size() << "\n";
        } else if(cmd == "empty") {
            cout << s.empty() << "\n";
        }
    }

    return 0;
}